File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        vector<int> v(n,1);
        for(int i = n-1;i>=0;i--){
            for(int j = i+1;j < n; j++){
                if(nums[i] < nums[j]){
                    v[i] = max(v[i], 1+v[j]);
                }
            }
        }
        return *max_element(v.begin(), v.end());
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-0.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int maxEl = 0;
        int maxRoom = 0;
        int result = 0;
        for(Interval interval: intervals){
            maxEl = Math.max(maxEl,(Math.max(interval.start,interval.end)));
            if(map.containsKey(interval.start)){
                map.put(interval.start, map.get(interval.start)+1);
            } else {
                map.put(interval.start, 1);
            }

            if(map.containsKey(interval.end)){
                map.put(interval.end, map.get(interval.end)-1);
            } else {
                map.put(interval.end, -1);
            }
        }
        for(int i = 0; i <= maxEl; i++){
            if(map.containsKey(i)){
                maxRoom += map.get(i);
                result = Math.max(result,maxRoom);
            }
        }
        return result;
    }
}
